packages/docs/site/docs/11-architecture/18-host-your-own-playground.md

```diff
@@ -0,0 +1,169 @@
+# Host your own Playground
+
+You can host the Playground on your own domain instead of `playground.wordpress.net`.
+
+This is useful for having full control over its content and behavior, as well as removing dependency on a third-party remote endpoint. It can provide a customized user experience, for example: preinstalled plugins, default theme, site settings, and demo content.
+
+Also, if you expect to see heavy traffic, it's good etiquette to provide your own server/CDN resources to serve the static file assets, rather than relying on the `wordpress.net` server.
+
+#### Usage
+
+A self-hosted Playground can be embedded as an iframe.
+
+```html
+<iframe src="https://my-playground.com"></iframe>
+```
+
+Or dynamically loaded by passing the remote URL to the Playground Client.
+
+```ts
+import { startPlaygroundWeb } from '@wp-playground/client';
+
+const client = await startPlaygroundWeb({
+	iframe: document.getElementById('wp'),
+	remoteUrl: `https://my-playground.com/remote.html`,
+});
+```
+
+## Build static assets
+
+Create a shallow clone of the Playground repository.
+
+```sh
+git clone -b trunk --single-branch --depth 1 git@github.com:WordPress/wordpress-playground.git
+```
+
+Install dependencies, and build the website.
+
+```sh
+npm install
+npm run build:website
+```
+
+This command internally runs the `nx` task `build:wasm-wordpress-net`. It copies the built assets from packages `remote` and `website` into a new folder at the path.
+
+```
+dist/packages/playground/wasm-wordpress-net
+```
+
+The entire service of the Playground consists of the content of this folder.
+
+It includes:
+
+-   Data and WASM files for all available PHP and WordPress versions
+-   `remote.html` - the core of Playground
+-   `index.html` - the shell, or browser chrome
+-   Web Worker script
+
+You can deploy the content of the folder to your server using SSH, such as `scp` or `rsync`.
+
+It is a static site, except for these dynamic aspects.
+
+-   Apache server directive `.htaccess` file - the result of combining `.htaccess` from the packages `remote` and `website`
+-   Plugin download proxy, `plugin-proxy.php`
+
+For these to work, you need a server environment with Apache and PHP installed.
+
+## NGINX configuration
+
+As an alternative to Apache, here is an example of using NGINX to serve the Playground.
+
+The Apache `.htaccess` file looks like this.
+
+```htaccess
+AddType application/wasm .wasm
+AddType	application/octet-stream .data
+
+<FilesMatch "iframe-worker.html$">
+  Header set Origin-Agent-Cluster: ?1
+</FilesMatch>
+
+Header set Cross-Origin-Resource-Policy: cross-origin
+Header set Cross-Origin-Embedder-Policy: credentialless
+RewriteEngine on
+RewriteRule ^plugin-proxy$ plugin-proxy.php [NC]
```

Comment:
I'm not sure why I put that rule in there. We could fetch() from `plugin-proxy.php` and ditch it. Still, it's needed at the moment.

Reply:
This part comes from the `packages/playground/website/public/.htaccess`. In fact, the entire file consists of these two lines.

Even if this part is removed from the `.htaccess`, its equivalent in NGINX is still necessary to selectively run PHP for that URL route (or all .php files).